module/reports/helper.js

```diff
@@ -170,154 +111,99 @@ module.exports = class ReportsHelper {
+								projectReport[project.status]++
 							}
 						}
-						//get total project count
 						projectReport['total'] =
 							projectReport[CONSTANTS.common.STARTED] +
 							projectReport['overdue'] +
 							projectReport[CONSTANTS.common.INPROGRESS_STATUS] +
 							projectReport[CONSTANTS.common.SUBMITTED_STATUS]
 
-						//Get tasks summary deatail of project.
 						if (project.taskReport) {
-							let keys = Object.keys(project.taskReport)
-							keys.map((key) => {
-								if (tasksReport[key]) {
-									tasksReport[key] = tasksReport[key] + project.taskReport[key]
-								} else {
-									tasksReport[key] = project.taskReport[key]
-								}
+							Object.keys(project.taskReport).forEach((key) => {
+								tasksReport[key] = (tasksReport[key] || 0) + project.taskReport[key]
 							})
 						}
 
-						//Get number of tasks overdued.
 						await Promise.all(
 							project.tasks.map((task) => {
-								//consider task only if not deleted
-								if (task.isDeleted == false && task.status != CONSTANTS.common.COMPLETED_STATUS) {
-									//Returns true or false
+								if (!task.isDeleted && task.status != CONSTANTS.common.COMPLETED_STATUS) {
 									let overdue = _getOverdueStatus(task.endDate)
-
 									if (overdue) {
-										if (tasksReport['overdue']) {
-											tasksReport['overdue'] = tasksReport['overdue'] + 1
-										} else {
-											tasksReport['overdue'] = 1
-										}
-										if (tasksReport[task.status]) {
-											tasksReport[task.status] = tasksReport[task.status] - 1
-										}
+										tasksReport['overdue'] = (tasksReport['overdue'] || 0) + 1
+										tasksReport[task.status]--
 									}
 								}
 							})
 						)
 					})
 				)
 
-				// if (UTILS.revertStatusorNot(appVersion)) {
-				// 	projectReport[CONSTANTS.common.COMPLETED_STATUS] = projectReport[CONSTANTS.common.SUBMITTED_STATUS]
-				// 	projectReport[CONSTANTS.common.NOT_STARTED_STATUS] = projectReport[CONSTANTS.common.STARTED]
-				// 	delete projectReport[CONSTANTS.common.SUBMITTED_STATUS]
-				// 	delete projectReport[CONSTANTS.common.STARTED]
-				// }
-
-				//Initially not supporting pdf report, hence resolving response here. Remove resolve if pdf capability needs to be added
-				let response = {
-					categories: categories,
-					tasks: tasksReport,
-					projects: projectReport,
-				}
-				return resolve({
-					success: true,
-					message: CONSTANTS.apiResponses.REPORTS_GENERATED,
-					data: {
-						dataAvailable: true,
-						data: response,
-					},
-				})
-
-				if (getPdf == true) {
+				if (getPdf) {
 					let reportTaskData = {}
-					Object.keys(tasksReport).map((taskData) => {
+					Object.keys(tasksReport).forEach((taskData) => {
 						reportTaskData[UTILS.camelCaseToTitleCase(taskData)] = tasksReport[taskData]
 					})
 
 					let categoryData = {}
-					Object.keys(categories).map((category) => {
+					Object.keys(categories).forEach((category) => {
```

Comment:
@Prajwal17Tunerlabs why map function changed to foreach

Reply:
so the data was not fetching correctly so i tried for each function but that was not the issue and i forgot to remove